### PR TITLE
#336: Correct month days in agenda

### PR DESCRIPTION
**Pull request by paulodamaso**
For #336: 

- Added the corresponding days for each month, including leap year calculations.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#351](https://github.com/zerocracy/datum/issues/351) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#351](https://github.com/zerocracy/datum/issues/351) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1680280555); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: xsl/templates.xsl [hunk 1]
```diff
@@ -46,7 +46,23 @@ SOFTWARE.
     <xsl:variable name="day" select="number(substring($iso, 9, 2))"/>
     <xsl:variable name="hour" select="number(substring($iso, 12, 2))"/>
     <xsl:variable name="minute" select="number(substring($iso, 15, 2))"/>
-    <xsl:value-of select="$minute + $hour * 60 + $day * 60 * 24 + $month * 60 * 24 * 30 + $year * 60 * 24 * 365"/>
+    <xsl:variable name="monthdays" select="28"/>
+    <xsl:choose>
+      <xsl:when 
+              test="$month = 1 or $month = 3 or $month = 5 or $month = 7 or $month = 8 or $month = 10 or $month = 12">
```
Comment (krzyk) [inline xsl/templates.xsl]: @paulodamaso I think this one should be indented by only 2 spaces more than the previous line

File: xsl/templates.xsl [hunk 2]
```diff
@@ -46,7 +46,23 @@ SOFTWARE.
...
+        <xsl:variable name="monthdays" select="31" />
+      </xsl:when>
+      <xsl:when 
+              test="$month = 4 or $month = 6 or $month = 9 or $month = 11">
```
Comment (krzyk) [inline xsl/templates.xsl]: @paulodamaso I think this one should be indented by only 2 spaces more than the previous line

File: xsl/templates.xsl [hunk 3]
```diff
@@ -46,7 +46,23 @@ SOFTWARE.
...
+        <xsl:variable name="monthdays" select="30" />
+      </xsl:when>
+      <xsl:when
+                test="$month = 2 and ($year mod 4 = 0 and $year mod 100 != 0) or $year mod 400">
```
Comment (krzyk) [inline xsl/templates.xsl]: @paulodamaso I think this one should be indented by only 2 spaces more than the previous line

File: xsl/templates.xsl [hunk 4]
```diff
@@ -46,7 +46,23 @@ SOFTWARE.
...
+        <xsl:variable name="monthdays" select="29" />
+      </xsl:when>
+    </xsl:choose>
+    <xsl:value-of
+            select="$minute + $hour * 60 + $day * 60 * 24 + $month * 60 * 24 * $monthdays + $year * 60 * 24 * 365"/>
```
Comment (krzyk) [inline xsl/templates.xsl]: @paulodamaso I think this one should be indented by only 2 spaces more than the previous line

**Comment by paulodamaso** [comment]
@krzyk Thanks for the tips, take a look now.

**Comment by krzyk** [comment]
@paulodamaso looks good

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
The job [#351](https://github.com/zerocracy/datum/issues/351) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@krzyk According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only cosmetic issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by krzyk** [comment]
@ypshenychka cosmetic, but failing the build so it couldn't be merged.
But yes, I confirm.

**Comment by ypshenychka** [comment]
@krzyk Thank you!

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**